owncloudApp/src/main/java/com/owncloud/android/presentation/adapters/transfers/TransfersAdapter.kt

```diff
@@ -0,0 +1,296 @@
+                    textViewFileCount.text = fileCountText
+
+                    when (headerItem.status) {
+                        TransferStatus.TRANSFER_FAILED -> {
+                            uploadListGroupButtonClear.apply {
+                                isVisible = true
+                                setOnClickListener {
+                                    clearFailed()
+                                }
+                            }
+                            uploadListGroupButtonRetry.apply {
+                                isVisible = true
+                                setOnClickListener {
+                                    retryFailed()
+                                }
+                            }
+                        }
+                        TransferStatus.TRANSFER_SUCCEEDED -> {
+                            uploadListGroupButtonClear.apply {
+                                isVisible = true
+                                setOnClickListener {
+                                    clearSuccessful()
+                                }
+                            }
+                            uploadListGroupButtonRetry.isVisible = false
+                        }
+                        else -> {
+                            uploadListGroupButtonRetry.isVisible = false
+                            uploadListGroupButtonClear.isVisible = false
+                        }
+                    }
+                }
+            }
+        }
+
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int, payloads: MutableList<Any>) {
+        if (payloads.isEmpty()) {
+            super.onBindViewHolder(holder, position, payloads)
+        } else {
+            (holder as TransferItemViewHolder).binding.apply {
+                if (payloads[0] is Int) {
+                    uploadProgressBar.progress = payloads[0] as Int
+                }
+            }
+        }
+
+    }
+
+    fun setData(transfers: List<OCTransfer>) {
+        val comparator: Comparator<OCTransfer> = object : Comparator<OCTransfer> {
+            override fun compare(transfer1: OCTransfer, transfer2: OCTransfer): Int {
+                return if (transfer1.transferEndTimestamp == null || transfer2.transferEndTimestamp == null) {
+                    compareUploadId(transfer1, transfer2)
+                } else {
+                    compareUpdateTime(transfer1, transfer2)
+                }
+            }
+
+            private fun compareUploadId(transfer1: OCTransfer, transfer2: OCTransfer): Int {
+                return (transfer1.id!!).compareTo(transfer2.id!!)
+            }
+
+            private fun compareUpdateTime(transfer1: OCTransfer, transfer2: OCTransfer): Int {
+                return (transfer2.transferEndTimestamp!!).compareTo(transfer1.transferEndTimestamp!!)
+            }
+        }
+
+        val transfersGroupedByStatus = transfers.groupBy { it.status }
```

Comment:
Yes, we can remove the comparator but I'll apply the sorting inside the iterator `forEach`, sorting the transfers from the same status group. At the end it will be the same result as the `groupBy` keeps the inorder, but this does exactly the same and I think it's more intuitive. Btw, in this case it should be `sortedByDescending` to keep the same order 👍